src/Workspaces/Core/Portable/Workspace/Solution/ProjectChanges.cs

```diff
@@ -116,6 +116,17 @@ public IEnumerable<DocumentId> GetAddedAdditionalDocuments()
             }
         }
 
+        public IEnumerable<DocumentId> GetAddedAnalyzerConfigDocuments()
+        {
+            foreach (var doc in _newProject.AnalyzerConfigDocuments)
+            {
+                if (!_oldProject.ContainsAnalyzerConfigDocument(doc.Id))
+                {
+                    yield return doc.Id;
+                }
+            }
+        }
```

Comment:
nit: `_newProject.AnalyzerConfigDocuments.Where(d => !oldProject.Contains(d.Id)).Select(d => d.Id);`?

Reply:
I have tried to retain the code style used in similar methods for existing document kinds :)

Reply:
Ah, i missed that.  Yes.  if we're already doing the copy/paste approach, we should likely continue with that.  Only tweak the methods if you're then going to share between the 3 impls :)